core/src/test/java/org/hibernate/ogm/test/hint/CustomHintSupportedProvider.java

```diff
@@ -0,0 +1,186 @@
+import org.hibernate.ogm.dialect.spi.NextValueRequest;
+import org.hibernate.ogm.dialect.spi.OperationContext;
+import org.hibernate.ogm.dialect.spi.TupleContext;
+import org.hibernate.ogm.dialect.spi.TupleTypeContext;
+import org.hibernate.ogm.entityentry.impl.TuplePointer;
+import org.hibernate.ogm.model.key.spi.AssociationKey;
+import org.hibernate.ogm.model.key.spi.AssociationKeyMetadata;
+import org.hibernate.ogm.model.key.spi.EntityKey;
+import org.hibernate.ogm.model.key.spi.EntityKeyMetadata;
+import org.hibernate.ogm.model.spi.Association;
+import org.hibernate.ogm.model.spi.Tuple;
+import org.hibernate.ogm.query.spi.QueryParserService;
+import org.hibernate.ogm.query.spi.QueryParsingResult;
+import org.hibernate.ogm.util.impl.CollectionHelper;
+import org.hibernate.query.internal.ParameterMetadataImpl;
+
+/**
+ * The provider using for test custom (non-standard) hint passing
+ * @author Sergey Chernolyas &amp;sergey_chernolyas@gmail.com&amp;
+ */
+public class CustomHintSupportedProvider extends BaseDatastoreProvider {
+
+	@Override
+	public Class<? extends GridDialect> getDefaultDialect() {
+		return CustomHintSupportedProvider.Dialect.class;
+	}
+
+	@Override
+	public Class<? extends QueryParserService> getDefaultQueryParserServiceType() {
+		return CustomHintSupportedParser.class;
+	}
+
+	public static class CustomHintSupportedParser implements QueryParserService {
+
+		@Override
+		public boolean supportsParameters() {
+			return false;
+		}
+
+		@Override
+		public QueryParsingResult parseQuery(SessionFactoryImplementor sessionFactory, String queryString, Map<String, Object> namedParameters) {
+			return new CustomHintSupportedQueryParsingResult();
+		}
+
+		@Override
+		public QueryParsingResult parseQuery(SessionFactoryImplementor sessionFactory, String queryString) {
+			return new CustomHintSupportedQueryParsingResult();
+		}
+	}
+
+	public static class CustomHintSupportedQueryParsingResult implements QueryParsingResult {
+
+		@Override
+		public Object getQueryObject() {
+			return "";
+		}
+
+		@Override
+		public List<String> getColumnNames() {
+			return Collections.emptyList();
+		}
+	}
+
+	public static class Dialect extends BaseGridDialect implements QueryableGridDialect<Serializable> {
+
+		public static final String DIALECT_SPECIFIED_HINT = Dialect.class.getName() + ".hint";
+
+		public Dialect(CustomHintSupportedProvider provider) {
+		}
+
+		@Override
+		public ClosableIterator<Tuple> executeBackendQuery(BackendQuery<Serializable> query, QueryParameters queryParameters, TupleContext tupleContext) {
+			List<String> queryHints = queryParameters.getQueryHints();
+			if ( queryHints.isEmpty() ) {
+				throw new HibernateException( "The query must have a hints!" );
+			}
+			else {
+				if ( !queryHints.contains( DIALECT_SPECIFIED_HINT ) ) {
+					throw new HibernateException( "The query not contains required hint! The query must contains hint \"" + DIALECT_SPECIFIED_HINT + "\"!" );
```

Comment:
`"The query does not contain the required hint \"" + DIALECT_SPECIFIED_HINT + "\"."`

Reply:
Ok.
I will fix.

Reply:
fixed